src/Docfx.Common/FileLinkInfo.cs

```diff
@@ -5,28 +5,28 @@
 
 namespace Docfx.Common;
 
-public struct FileLinkInfo
+public readonly struct FileLinkInfo
```

Comment:
It might be better to just turn this into a class, because the struct would be boxed when calling `GenerateHref(IFileLinkInfo href);`
```suggestion
public class FileLinkInfo
```

Reply:
I've committed struct -> class changes. and relevant test code. 

> IIRC some methods create the FileLinkInfo instance only if there is an ICustomHrefGenerator, but others don't check.

It seems `FileLinkInfo`  class is used for calculatating Href.